### PR TITLE
Windows management for the render process

### PR DESCRIPTION
**Pull request by chicoxyzzy**
- make all windows management messages resolve in a proper window (the one they was called from, see [io changes](https://github.com/socketsupply/io/pull/44))
- allow broadcasting to all the windows via `ssc-node`'s send
- remove `getConfig` message handler from the render process (all necessary config fields are available via `window.__args` or `io.runtime.args`)

### CODE REVIEW

File: src/desktop/main.cc [hunk 1]
```diff
@@ -353,6 +375,29 @@ MAIN {
...
+          auto window = windowFactory.getWindow(message.index);
+          if (window) {
+            window->eval(getEmitToRenderProcessJavaScript(
+              decodeURIComponent(message.get("event")),
+              value
+            ));
+          }
+        } else {
+          for (auto w : windowFactory.windows) {
+            if (w != nullptr) {
```
Comment (jwerle) [inline src/desktop/main.cc:389]: `w` is a `Window*` so no need to call `getWindow()` below as `window == w` here

Comment (chicoxyzzy) [reply src/desktop/main.cc:389]: > `w` is a `Window*` so no need to call `getWindow()` below as `window == w` here

it was `WindowWithMetadate*`

File: src/core/android.cc [hunk 2]
```diff
@@ -233,10 +233,7 @@ jboolean NativeCore::ConfigureEnvironment () {
     }
   }
 
-  windowOptions.executable = this->config["executable"];
   windowOptions.headless = this->config["headless"] == "true";
-  windowOptions.version = "v" + this->config["version"];
-  windowOptions.title = this->config["title"];
```
Comment (chicoxyzzy) [inline src/core/android.cc:239]: this 3 are in the config already

File: src/desktop/main.cc [hunk 3]
```diff
@@ -433,12 +476,12 @@ MAIN {
           indexMain = std::stoi(message.get("indexMain"));
           indexSub = std::stoi(message.get("indexSub"));
         } catch (...) {
-          window->resolvePromise(seq, OK_STATE, "");
```
Comment (chicoxyzzy) [inline src/desktop/main.cc:436]: this was failing to parse

File: src/desktop/main.cc [hunk 4]
```diff
@@ -674,15 +777,14 @@ MAIN {
       return;
     }
 
-    if (message.name == "getConfig") {
-      const auto seq = message.get("seq");
-      auto wrapped = ("\"" + String(_settings) + "\"");
-      window->resolvePromise(seq, OK_STATE, encodeURIComponent(wrapped));
-      return;
-    }
```
Comment (chicoxyzzy) [inline src/desktop/main.cc:682]: we have config in the `window.__args.config` / `io.runtime.args`

**Review by jwerle — APPROVED** [review]

**Pull request merged**